source/ts/poll.ts

```diff
@@ -20,10 +23,26 @@ export async function pollUntil<T>(
           return;
         }
 
-        query();
+        queryTimer = setTimeout(query, 0);
       });
     };
 
     query();
   });
 }
+
+export async function waitForResultToBe<T>(
```

Comment:
I wonder if this should just be `waitForResult`

Reply:
☝🏼  optional. I'd happily approve as-is

Reply:
Or possibly, `pollUntilResult` 🤔

Reply:
Went with `waitForResult`, you could argue that the polling is implementation detail